enumerable.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    my_each { |e| temp << e if yield(e) }
+    temp
+  end
```

Comment:
In `my_select`
- enumerables my_select returns an enumerator if no block is given